File: LeetCode/0747. Largest Number At Least Twice of Others/solution.cpp
/**
 * 232 / 232 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 10.5 MB 
 */
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        if (nums.size() == 1) return 0;
        int first = 0, second = -1;
        for (int i = 1; i < nums.size(); ++ i) {
            if (nums[i] > nums[first]) {
                second = first;
                first = i;
            } else if (second == -1 || nums[i] > nums[second]) {
                second = i;
            }
        }
        if (nums[first] >> 1 >= nums[second]) {
            return first;
        }
        return -1;
    }
};

File: LeetCode/0747. Largest Number At Least Twice of Others/solution.py
"""
232 / 232 test cases passed.
Runtime: 36 ms
Memory Usage: 15 MB
"""
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return 0
        first, second = 0, -1
        for i in range(1, n):
            if nums[i] > nums[first]:
                second = first
                first = i
            elif second == -1 or nums[i] > nums[second]:
                second = i
        return first if nums[first] >> 1 >= nums[second] else -1
